shell/platform/darwin/macos/framework/Source/FlutterView.mm

```diff
@@ -20,14 +33,73 @@ - (instancetype)initWithFrame:(NSRect)frame
   if (self) {
     self.openGLContext = [[NSOpenGLContext alloc] initWithFormat:shareContext.pixelFormat
                                                     shareContext:shareContext];
+
+    [self setWantsLayer:YES];
+
+    // Layer for content (which will be set by surfaceManager). This is separate from
+    // self.layer because it needs to be flipped vertically (using layer.sublayerTransform)
+    contentLayer = [[CALayer alloc] init];
+    [self.layer addSublayer:contentLayer];
+
+    resizeSynchronizer = [[FlutterResizeSynchronizer alloc] initWithDelegate:self];
+    surfaceManager = [[FlutterSurfaceManager alloc] initWithLayer:contentLayer
+                                                    openGLContext:self.openGLContext];
+
     _reshapeListener = reshapeListener;
-    self.wantsBestResolutionOpenGLSurface = YES;
   }
   return self;
 }
 
+- (void)resizeSynchronizerCommit:(FlutterResizeSynchronizer*)synchronizer {
+  [self.openGLContext makeCurrentContext];
+  glFlush();
+  [NSOpenGLContext clearCurrentContext];
+
+  [CATransaction begin];
+  [CATransaction setDisableActions:YES];
+  self.layer.frame = self.bounds;
+  self.layer.sublayerTransform = CATransform3DTranslate(CATransform3DMakeScale(1, -1, 1), 0,
+                                                        -self.layer.bounds.size.height, 0);
+  contentLayer.frame = self.layer.bounds;
+
+  [surfaceManager swapBuffers];
+
+  [CATransaction commit];
+}
+
+- (int)getFrameBufferIdForSize:(CGSize)size {
+  if ([resizeSynchronizer shouldEnsureSurfaceForSize:size]) {
+    [surfaceManager ensureSurfaceSize:size];
+  }
+  return [surfaceManager glFrameBufferId];
+}
+
+- (void)present {
+  [resizeSynchronizer requestCommit];
+}
+
+- (void)start {
```

Comment:
I'm still not sure if `start` is something we want outside of the `init` method. Are there situations where we will want to create the `FlutterView` and not `start` it right away?

Reply:
The problem here is that we block the platform thread every time the view is resized, which is detected by overriding `[NSView setFrameSize]`. It is normal to get some calls to `setFrameSize` before the shell is running and producing frames. If we block the thread too soon, it will deadlock. So `[FlutterView start]` basically tells the view that the shell is ready and running and that if it block the platform thread, a frame will come.

Reply:
So is this call more like: `readyForSynchronousResizing`?

Reply:
I changed it [here](https://github.com/flutter/engine/pull/21525/commits/c32f0cd65aceb54d688363b92e38ff548b256b97). `[FlutterView start]` is gone, instead there is `synchronousResizing` property, which when true means the view will block the platform thread during resizing. If false, resizing still works in a non blocking way.